service/controller/network/v1/resource/ipam/create.go

```diff
@@ -1,9 +1,62 @@
 package ipam
 
-import "context"
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/meta"
+
+	"github.com/giantswarm/cluster-operator/service/controller/network/v1/key"
+)
 
 // EnsureCreated takes care of cluster subnet allocation when
 // ClusterNetworkConfig object is created.
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	clusterNetworkCfg, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	clusterID := key.ClusterID(clusterNetworkCfg)
+
+	if clusterNetworkCfg.Status.IP != "" {
+		// Subnet allocated. No need to do anything.
+		r.logger.LogCtx(ctx, "level", "debug", "message", "Subnet allocated. No need to do anything.", "clusterID", clusterID)
```

Comment:
I would not like to start using sentences in logs. A short and crispy message should be to the point. Multiple sentences usually point into different semantic directions. We should then rather emit two log lines IMO. Further, the usual pattern so far was to mention that there was something and due to this something the reconciliation gets canceled for the resource. Check the other projects. We have plenty of occasions for this.

Reply:
Multiple log lines are bad thing. In general it's impossible to connect two log lines with 100% accuracy when there's no trace cookie present (and we don't yet). I'll try to rephrase these.